### PR TITLE
dev/core#1936 Make the label field on price_field_value non required …

### PR DESCRIPTION
**Pull request by seamuslee001**
…field

Overview
----------------------------------------
5.27 port of https://github.com/civicrm/civicrm-core/pull/18123

Before
----------------------------------------
Label field required

After
----------------------------------------
label field non required

ping @KarinG @demeritcowboy @eileenmcnaughton 

Karin in your testing you should update the upgrade function to be 5_27_4 but we need 5_27_5 for core because that will be the next available version

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FiveTwentySeven.php [hunk 1]
```diff
@@ -74,6 +74,35 @@ public function upgrade_5_27_alpha1($rev) {
...
+    $locales = CRM_Core_I18n::getMultilingual();
+    if ($locales) {
+      foreach ($locales as $locale) {
+        CRM_Core_DAO::executeQuery("UPDATE civicrm_price_field_value SET label_{$locale} = NULL WHERE label_{$locale} = 'null'", [], TRUE, NULL, FALSE, FALSE);
+        CRM_Core_DAO::executeQuery("ALTER TABLE civicrm_price_field_value CHANGE `label_{$locale}` `label_{$locale}` varchar(255) DEFAULT NULL  COMMENT 'Price field option label'", [], TRUE, NULL, FALSE, FALSE);
+      }
+    }
+    else {
+      CRM_Core_DAO::executeQuery("UPDATE civicrm_price_field_value SET label = NULL WHERE label = 'null'", [], TRUE, NULL, FALSE, FALSE);
+      CRM_Core_DAO::executeQuery("ALTER TABLE civicrm_price_field_value CHANGE `label` `label` varchar(255) DEFAULT NULL  COMMENT 'Price field option label'", [], TRUE, NULL, FALSE, FALSE);
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/php/FiveTwentySeven.php:101]: These lines should be in reverse order. At the time the first one runs the column is not null so can't set it to null. Ditto for locales above.

Comment (KarinG) [reply CRM/Upgrade/Incremental/php/FiveTwentySeven.php:101]: That explains a lot! I’ll run another full-w/ upgrade test swapping these.

**Comment by KarinG** [comment]
I can not upgrade -> I still get stuck on [Executed: Upgrade DB to 5.27.4: SQL]

`[11-Aug-2020 10:40:48 America/Edmonton] Error: Call to undefined method CRM_Core_I18n::getMultilingual() in /Applications/MAMP/htdocs/d8civicrm.local/vendor/civicrm/civicrm-core/CRM/Upgrade/Incremental/php/FiveTwentySeven.php on line 92 #0 /Applications/MAMP/htdocs/d8civicrm.local/vendor/civicrm/civicrm-core/CRM/Queue/Task.php(74): CRM_Upgrade_Incremental_php_FiveTwentySeven::priceFieldValueLabelNonRequired(Object(CRM_Queue_TaskContext))`

**Comment by KarinG** [comment]
Re-ran it with xdebug: 
image

**Comment by KarinG** [comment]
Notes $locales are fetched differently for `public function priceFieldValueLabelRequired($ctx) {`

```
    $domain = new CRM_Core_DAO_Domain();
    $domain->find(TRUE);
    if ($domain->locales) {
      $locales = explode(CRM_Core_DAO::VALUE_SEPARATOR, $domain->locales);
```

**Comment by seamuslee001** [comment]
thanks @demeritcowboy @KarinG fixed the upgrader now

File: CRM/Upgrade/Incremental/php/FiveTwentySeven.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FiveTwentySeven.php:101]: This has been resolved now

**Comment by demeritcowboy** [comment]
I think you need to explode/unserialize the locales as @karing notes, as in CRM_Core_I18n::getMultilingual()

**Comment by seamuslee001** [comment]
thanks @demeritcowboy fixed now

**Comment by KarinG** [comment]
Ok one more time! :-)

Installed 5.27.4 - vanilla - create Contribution Page -> reproducing the issue:
image
image

Ok then:

karins-MBP:civicrm-core sysadmin$ curl https://patch-diff.githubusercontent.com/raw/civicrm/civicrm-core/pull/18124.diff -o 18124.diff

karins-MBP:civicrm-core sysadmin$ cat 18124.diff |patch -p1
patching file CRM/Core/I18n/SchemaStructure.php
patching file CRM/Price/DAO/PriceFieldValue.php
patching file CRM/Upgrade/Incremental/php/FiveTwentySeven.php
patching file Civi/Api4/Service/Spec/Provider/PriceFieldValueCreationSpecProvider.php
patching file tests/phpunit/CRM/Price/BAO/PriceFieldValueTest.php
patching file xml/schema/Price/PriceFieldValue.xml
karins-MBP:civicrm-core sysadmin$ 

For testing the upgrade
set civicrm_domain.version = 5.27.3

emacs CRM/Upgrade/Incremental/php/FiveTwentySeven.php
change -> public function upgrade_5_27_5($rev) {
to: public function upgrade_5_27_4($rev) {

Ok here we go - running upgrade -> Done!

Checking:

This looks good:
image

and this looks good too:
image

**Comment by seamuslee001** [comment]
going to add Merge on pass based on @KarinG 's testing on the linked PRs

**Comment by seamuslee001** [comment]
Test fails unrelated

**Comment by KarinG** [comment]
Phew!

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001  - the approach had been agreed on chat & I agree @KarinG's testing was solid on this

**Pull request merged**